### PR TITLE
py3.10 compatibility

### LINKED ISSUE #11: Py 3.10 compatibility
From 3.10 [Changelogs](https://docs.python.org/3/whatsnew/3.10.html#argparse): 

argparse
Misleading phrase “optional arguments” was replaced with “options” in argparse help. Some tests might require adaptation if they rely on exact output match. (Contributed by Raymond Hettinger in [bpo-9694](https://bugs.python.org/issue?@action=redirect&bpo=9694).)

### PR DESCRIPTION
**Pull request by ig0rsky**
Fixes: #11

* https://github.com/python/cpython/commit/fb35fa49d192368e94ffec09c092260ed0fea2e1
* https://github.com/python/cpython/commit/41b223d29cdfeb1f222c12c3abaccc3bc128f5e7

### CODE REVIEW

**Comment by ig0rsky** [comment]
tested with a few of the examples, seems to work fine. 
should we maintain backwards compatibility, or increase major v. to 1?

**Comment by kootenpv** [comment]
We definitely should maintain backwards compatibility, so I guess we should do a check for which argparse version is being used?

```python
pascal@x1x:/home/pascal$ ipython
Python 3.9.6 | packaged by conda-forge | (default, Jul 11 2021, 03:39:48) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.0.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import argparse

In [2]: argparse.__version__
Out[2]: '1.1'
```

what's yours?

**Comment by ig0rsky** [comment]
```
ipython                                                                                                                                                                                           (mgnr-mamba-macos) 
Python 3.10.4 (main, Mar 31 2022, 03:38:35) [Clang 12.0.0 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.3.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import argparse

In [2]: argparse.__version__
Out[2]: '1.1'
```

File: cliche/argparser.py [hunk 1]
```diff
@@ -14,6 +14,13 @@
 container_fn_name_to_type = {}
 class_init_lookup = {}  # for class functions
 
+def _check_py_version(new, old):
+    import sys
+    if sys.version_info.major >= 3 and sys.version_info.minor >= 10:
+        return new
+    else:
+        return old
+
```
Comment (kootenpv) [inline cliche/argparser.py]: Just make this a constant:

```python
PYTHON_310_OR_UP = sys.version_info >= (3, 10)
```

File: cliche/argparser.py [hunk 2]
```diff
@@ -78,7 +85,7 @@ def _print_message(self, message, file=None, color=None):
 
                     message = self.make_subgroups(message)
 
-                    message = message.replace("optional arguments:", "OPTIONAL ARGUMENTS:")
+                    message = _check_py_version( message.replace("options", "OPTIONS"), message.replace("optional arguments", "OPTIONAL ARGUMENTS"))
```
Comment (kootenpv) [inline cliche/argparser.py]:
```python
message = message.replace("options" if PYTHON_310_OR_UP else "optional arguments", "OPTIONS:")
```

File: cliche/argparser.py [hunk 3]
```diff
@@ -64,7 +71,7 @@ def _print_message(self, message, file=None, color=None):
                             ms = ms[0]
                             first_start = message.index("positional arguments")
                             start = first_start + message[first_start:].index(ms) + len(ms)
-                            end = message.index("optional ")
+                            end = _check_py_version(message.index("options"), message.index("optional"))
```
Comment (kootenpv) [inline cliche/argparser.py]:
```python
end = message.index("options " if PYTHON_310_OR_UP else "optional ")
```

File: cliche/argparser.py [hunk 4]
```diff
@@ -123,7 +130,8 @@ def error(self, message):
         # otherwise it prints generic help but it should print the specific help of the subcommand
         if "unrecognized arguments" in message:
             multiple_args = message.count(" ") > 2
-            type_arg_msg = "Unknown optional argument" if "-" in message else "Extra positional argument"
+            option_str = _check_py_version("Unknown option", "Unknown optional argument")
```
Comment (kootenpv) [inline cliche/argparser.py]:
```python
option_str = "Unknown option" if PYTHON_310_OR_UP else "Unknown optional argument"
```

File: setup.py [hunk 5]
```diff
@@ -2,8 +2,8 @@
 from setuptools import setup
 
 MAJOR_VERSION = "0"
-MINOR_VERSION = "10"
-MICRO_VERSION = "95"
+MINOR_VERSION = "11"
+MICRO_VERSION = "0"
 VERSION = "{}.{}.{}".format(MAJOR_VERSION, MINOR_VERSION, MICRO_VERSION)
```
Comment (kootenpv) [inline setup.py]: can leave it at minor=10 as from the perspective of cliche there is no significant/breaking change yet

**Review by kootenpv — CHANGES_REQUESTED** [review]
thanks, some small suggestions!

**Comment by kootenpv** [comment]
cool thanks, will test locally and upload to pypi

**Pull request merged**